torch/_inductor/rocm_multiarch_utils.py

```diff
@@ -0,0 +1,301 @@
+"""
+
+import logging
+import os
+import subprocess
+from typing import Optional
+
+from torch.utils.cpp_extension import _join_rocm_home, ROCM_HOME
+
+
+log = logging.getLogger(__name__)
+
+
+def get_rocm_compiler() -> str:
+    """
+    Get path to ROCm's clang compiler.
+    Uses PyTorch's ROCM_HOME detection.
+
+    Returns:
+        Path to clang compiler
+
+    Raises:
+        RuntimeError: If ROCm is not found
+    """
+    if ROCM_HOME is None:
+        raise RuntimeError(
+            "ROCm installation not found. "
+            "PyTorch was not built with ROCm support or ROCM_HOME is not set."
+        )
+
+    # ROCm's clang is at <ROCM_HOME>/llvm/bin/clang
+    clang_path = _join_rocm_home("llvm", "bin", "clang")
+
+    if not os.path.exists(clang_path):
+        raise RuntimeError(
+            f"ROCm clang not found at {clang_path}. ROCM_HOME is set to {ROCM_HOME}"
+        )
+
+    return clang_path
+
+
+def get_rocm_bundler() -> str:
+    """
+    Get path to clang-offload-bundler.
+    Uses PyTorch's ROCM_HOME detection.
+
+    Returns:
+        Path to bundler
+
+    Raises:
+        RuntimeError: If bundler is not found
+    """
+    if ROCM_HOME is None:
+        raise RuntimeError(
+            "ROCm installation not found. "
+            "PyTorch was not built with ROCm support or ROCM_HOME is not set."
+        )
+
+    # Bundler is at <ROCM_HOME>/llvm/bin/clang-offload-bundler
+    bundler_path = _join_rocm_home("llvm", "bin", "clang-offload-bundler")
+
+    if not os.path.exists(bundler_path):
+        raise RuntimeError(
+            f"clang-offload-bundler not found at {bundler_path}. "
+            f"ROCM_HOME is set to {ROCM_HOME}"
+        )
+
+    return bundler_path
+
+
+def get_rocm_target_archs() -> list[str]:
+    """
+    Get target architectures from environment or config.
+    Returns: List of architecture strings (e.g., ['gfx90a', 'gfx942'])
+    """
+    # Check PYTORCH_ROCM_ARCH environment variable
+    env_archs = os.environ.get("PYTORCH_ROCM_ARCH", "").strip()
+    if env_archs:
+        archs = [arch.strip() for arch in env_archs.replace(";", ",").split(",")]
+        archs = [arch for arch in archs if arch]
+        if archs:
+            log.info(f"Using ROCm architectures from PYTORCH_ROCM_ARCH: {archs}")
+            return archs
+
+    # Try to get from inductor config
+    try:
+        from torch._inductor import config
+
+        if hasattr(config, "rocm") and hasattr(config.rocm, "target_archs"):
+            archs = config.rocm.target_archs
+            if archs:
+                log.info(f"Using ROCm architectures from config: {archs}")
+                return archs
+    except Exception as e:
+        log.debug(f"Could not read config.rocm.target_archs: {e}")
+
+    # Default to common MI300/MI450 architectures
+    default_archs = ["gfx90a", "gfx942", "gfx1100", "gfx1101"]
+    log.info(f"Using default ROCm architectures: {default_archs}")
+    return default_archs
```

Comment:
When torch is built, it saves the gfx targets used for the build.

```suggestion
    return torch.cuda.get_arch_list()
```